src/core/class/mediarithmics/ad-renderer/AdRendererTemplatePlugin.ts

```diff
@@ -131,7 +127,19 @@ export abstract class AdRendererTemplatePlugin extends AdRendererBasePlugin<
     const compiledTemplate = this.engineBuilder.compile(template);
 
     const creativeClickUrl =
-      urlProperty && urlProperty.value.url ? urlProperty.value.url : undefined;
+      urlProperty && urlProperty.value && urlProperty.value.url
+        ? urlProperty.value.url
+        : undefined;
+
+    if (
+      urlProperty &&
+      urlProperty.value &&
+      (!urlProperty.value.url || urlProperty.value.url.length === 0)
+    ) {
+      const msg = `crid: ${creativeId} - Url property is undefined`;
```

Comment:
Did that change anything ? You will still print a warning when building an instance context.
Also, a better message (explaining why it may be a bad thing and when it could be ok) may help.